src/homework2.ts

```diff
@@ -0,0 +1,163 @@
+// Create a function which uses tuple type to calculate the distance between two points in 2D space
+
+function excercise4() {
+    let point1:[number, number] = [1, 1]
+    let point2:[number, number] = [4, 5]
+
+    function distance(p1:[number, number], p2:[number, number]): number {
+        const x1 = p1[0];
+        const y1 = p1[1];
+        const x2 = p2[0];
+        const y2 = p2[1];
+
+        const d:number = Math.sqrt((x2 - x1)**2 + (y2 - y1)**2);
+        return d;
+    }
+    console.log(`Excercise4: distance = ${distance(point1, point2)}`);
+}
+excercise4();
+
+// Create a function which uses type alias to calculate the distance between two points in 2D 
+// space - points are objects with x and y properties
+
+function excercise5() {
+    type TPoint = {
+        x: number;
+        y: number;
+    }
+
+    let point1:TPoint = {x:1, y:1};
+    let point2:TPoint = {x:4, y:5};
+
+    function distance(p1: TPoint, p2: TPoint): number {
+        const x1 = p1.x;
```

Comment:
// TODO: use distructuring to get x and y from p1 and p2
this  requirement is not covered in the code